core/trino-main/src/main/java/io/trino/server/InternalAuthenticationManager.java

```diff
@@ -118,17 +125,26 @@ public void handleInternalRequest(ContainerRequestContext request)
     public Request filterRequest(Request request)
     {
         return fromRequest(request)
-                .addHeader(TRINO_INTERNAL_BEARER, generateJwt())
+                .addHeader(TRINO_INTERNAL_BEARER, getOrGenerateJwt())
                 .build();
     }
 
-    private String generateJwt()
+    private synchronized String getOrGenerateJwt()
```

Comment:
this is at cost of synchronization. Not sure worth it.

...

Actually I think this is wrong. With this change you can use a token which is "almost" expired. Risking communicatino failure. If you really need this change (which I think you do not) you should refresh token significantly earlier than expiration time. Maybe half way.